app/bundles/NotificationBundle/Helper/NotificationUploader.php

```diff
@@ -0,0 +1,145 @@
+    {
+        $files         = $request->files->all()['notification'];
+
+        $uploadedFiles = [];
+        $deteledFiles  = [];
+        foreach ($this->getUploadFilesName() as $fileName) {
+            $uploadDir    = $this->getUploadDir($notification);
+
+            // Delete file
+            if (!empty($form->get($fileName.'_delete')->getData())) {
+                $this->fileUploader->delete($uploadDir.DIRECTORY_SEPARATOR.$this->getEntityVar($notification, $fileName));
+                $this->getEntityVar($notification, $fileName, 'set', '');
+            }
+
+            /* @var UploadedFile $file */
+            if (empty($files[$fileName])) {
+                continue;
+            }
+            $file = $files[$fileName];
+
+            try {
+                $uploadedFile = $this->fileUploader->upload($uploadDir, $file);
+                $this->getEntityVar($notification, $fileName, 'set', $uploadedFile);
+            } catch (FileUploadException $e) {
+                $this->fileUploader->delete($uploadDir.DIRECTORY_SEPARATOR.$uploadedFile);
+            }
+        }
+    }
+
+    /**
+     * @param Notification $notification
+     * @param string       $fileName
+     *
+     * @return string
+     */
+    public function getCompleteFilePath(Notification $notification, $fileName)
+    {
+        $uploadDir = $this->getUploadDir($notification);
+
+        return $uploadDir.DIRECTORY_SEPARATOR.$fileName;
+    }
+
+    /**
+     * @param Notification $notification
+     */
+    public function deleteAllFilesOfNotification(Notification $notification)
+    {
+        $uploadDir = $this->getUploadDir($notification);
+        $this->fileUploader->delete($uploadDir);
+    }
+
+    /**
+     * @param $notification
+     *
+     * @return string
+     */
+    public function getFullUrl($notification, $key)
+    {
+        if ($fileName = $this->getEntityVar($notification, $key)) {
+            return $this->coreParametersHelper->getParameter('site_url').DIRECTORY_SEPARATOR.$this->coreParametersHelper->getParameter('notification_upload_dir').DIRECTORY_SEPARATOR.$notification->getId().DIRECTORY_SEPARATOR.$fileName;
+        }
+    }
+
+    /**
+     * @param object $entity
+     * @param string $key
+     * @param string $action
+     */
+    public function getEntityVar($entity, $key, $action = 'get', $value = '')
+    {
+        $var = $action.ucfirst($key);
+        if ($action == 'get') {
+            return $entity->$var();
+        } else {
+            $entity->$var((string) $value);
+        }
+    }
+
+    /**
+     * @param Notification $notification
+     *
+     * @return string
+     */
+    private function getUploadDir(Notification $notification)
+    {
+        return $this->coreParametersHelper->getParameter('kernel.root_dir').DIRECTORY_SEPARATOR.'..'.DIRECTORY_SEPARATOR.$this->coreParametersHelper->getParameter('notification_upload_dir').DIRECTORY_SEPARATOR.$notification->getId();
```

Comment:
This won't work for all setups.  notification_upload_dir should be configurable to an absolute path. Let's default it to `%kernel.root_dir%/media/images/notifications` in the config.php changing this line to `return $this->coreParametersHelper->getParameter('notification_upload_dir').DIRECTORY_SEPARATOR.$notification->getId();`

Reply:
Actually, is there a need for a separate directory than just using our existing image_path? If not, let's just use relative image_path and then use the PathsHelper to get the uploadDir.

Reply:
@alanhartless do you wanna store all images to one directory. Thousands of images in one directory is hell.

Reply:
No, you can still put it in a hard coded subdirectory in the images folder.